share/models/change.py

```diff
@@ -89,7 +93,7 @@ def accept(self, save=True):
         ret = []
         with transaction.atomic():
             self._changes_cache = list(self.changes.all())
-            for c in self._changes_cache:
+            for c in sorted(self._changes_cache, key=lambda c: c.type):
```

Comment:
This feels a bit scary. The order of the changes does matter IIRC. If it keeps the order within the types, you're probably good.